funding/manager_test.go

```diff
@@ -4444,3 +4444,35 @@ func TestFundingManagerZeroConf(t *testing.T) {
 	// have been deleted from the database, as the channel is announced.
 	assertNoFwdingPolicy(t, alice, bob, fundingOp)
 }
+
+// TestCommitmentTypeFundmaxSanityCheck was introduced as a way of reminding
+// developers of new channel commitment types to also consider the channel
+// opening behavior with a specified fundmax flag. To give a hypothetical
+// example, if ANCHOR types had been introduced after the fundmax flag had been
+// activated, the developer would have had to code for the anchor reserve in the
+// funding manager in the context of public and private channels. Otherwise
+// inconsisten bahvior would have resulted when specifying fundmax for different
+// types of channel openings.
+// To ensure consistency this test compares a map of locally defined channel
+// commitment types to the list of channel types that are defined in the proto
+// files. It fails if the proto files contain additional commitment types. Once
+// the developer considered the new channel type behavior it can be added in
+// this test to the map `allCommitmentTypes`.
+func TestCommitmentTypeFundmaxSanityCheck(t *testing.T) {
```

Comment:
I don't know if this test is that useful?

Reply:
I added this test-case as it was envisaged in the original PR: https://github.com/lightningnetwork/lnd/pull/4029#issuecomment-1000242502

As the test documentation lays out it helps developers of new commitment types be reminded to consider the 'fundmax' behavior of their implementation, e.g. provide additional reserves or similar constraints.

Does that make sense?

Reply:
I think its a good idea to have such a check, or we include a reserve for all types except the one we no don't need one ?